src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java

```diff
@@ -0,0 +1,142 @@
+        Order order = gson.fromJson(outputGetBasket, Order.class);
+
+        // Assertions to validate basket content
+        Assertions.assertEquals("testuserid", order.getBuyer(), "The user id doesn't match");
+        Assertions.assertEquals(2, order.getOrderItems().size(), "More than 2 items were found in the order");
+
+        List<OrderItem> listitems = order.getOrderItems();
+        // Assertions for each order item
+        Assertions.assertEquals(5, listitems.get(0).getProductId());
+        Assertions.assertEquals("Roslyn Red Pin", listitems.get(0).getProductName());
+        Assertions.assertEquals(2, listitems.get(0).getUnits());
+        Assertions.assertEquals(8.5, listitems.get(0).getUnitPrice());
+
+        Assertions.assertEquals(3, listitems.get(1).getProductId());
+        Assertions.assertEquals("Prism White T-Shirt", listitems.get(1).getProductName());
+        Assertions.assertEquals(1, listitems.get(1).getUnits());
+        Assertions.assertEquals(12, listitems.get(1).getUnitPrice());
+    }
+
+    // Method to add items to the basket
+    public String addItemsToBasket() throws IOException {
+        // Debug log for creating connection
+        log.debug("Creating the connection with URL: http://basket_api_" + tJobName + ":80/api/v1/Basket");
+        // Create HTTP client and POST request
+        HttpClient httpclient = HttpClients.createDefault();
+        String requestURL ="http://webshoppingagg_" + tJobName + ":80/api/v1/Basket";
+        //requestURL = "http://localhost:5016//api/v1/Basket";
+        HttpPost httpPost = new HttpPost(requestURL);
+        // Configure headers
+        httpPost.addHeader("accept", "text/plain");
+        httpPost.addHeader("content-type", "application/json");
+        httpPost.addHeader("Authorization", "Bearer " + tokenAPI);
+
+        // JSON payload for adding items to the basket
+        String json = "{\n" +
+                "  \"buyerId\": \"testuserid\",\n" +
+                "  \"items\": [\n" +
+                "    {\n" +
+                "      \"id\": \"testproductid1\",\n" +
+                "      \"productId\": \"5\",\n" +
+                "      \"quantity\": \"2\"\n" +
+                "    },\n" +
+                "    {\n" +
+                "      \"id\": \"testproductid2\",\n" +
+                "      \"productId\": \"3\",\n" +
+                "      \"quantity\": \"1\"\n" +
+                "    }\n" +
+                "\n" +
+                "  ]\n" +
+                "}";
+
+        // Set JSON payload as entity for the HTTP request
+        StringEntity entity = new StringEntity(json);
+        httpPost.setEntity(entity);
+
+        // Define response handler
+        ResponseHandler<String> responseHandler = new BasicResponseHandler();
+        // Debug log for performing the request
+        log.debug("Performing the request");
+        // Execute HTTP request and return response
+        return httpclient.execute(httpPost, responseHandler);
+    }
```

Comment:
This method and the next have the same problem as the base class, there are many trivial comments